src/prototype_room_market.js

```diff
@@ -169,15 +175,41 @@ Room.prototype.sendPowerOwnRooms = function() {
   }
 };
 
+Room.prototype.sendEnergyToMyRooms = function() {
+  if ((_.size(Memory.needEnergyRooms) > 0) && (_.size(Memory.canHelpRooms) > 0) &&
+    _.includes(Memory.canHelpRooms, this.name) && !_.includes(Memory.needEnergyRooms, this.name) &&
+    (this.terminal.store[RESOURCE_ENERGY] > config.terminal.minEnergyAmount)) {
+    const diffNeedEnergyRooms = _.difference(Memory.needEnergyRooms, Memory.canHelpRooms);
+    const needEnergyRooms = _.size(diffNeedEnergyRooms) > 0 ? diffNeedEnergyRooms : Memory.needEnergyRooms;
+    const myRoom = _.shuffle(needEnergyRooms)[0];
+    if (myRoom) {
+      const amount = (this.terminal.store[RESOURCE_ENERGY] - config.terminal.minEnergyAmount) * 100;
+      if ((amount > 100) && (amount < this.terminal.store[RESOURCE_ENERGY]) &&
+        Game.rooms[myRoom].terminal &&
+        (Game.rooms[myRoom].terminal.store[RESOURCE_ENERGY] < config.terminal.maxEnergyAmount)) {
+        const success = this.terminal.send(RESOURCE_ENERGY, amount, myRoom, 'send energy ' + this.name + ' ' + myRoom) === OK;
+        if (success) {
+          const cost = Game.market.calcTransactionCost(amount, this.name, myRoom);
+          this.log('sendEnergyToMyRooms', myRoom, amount, cost, this.terminal.store[RESOURCE_ENERGY]);
+        }
+      }
+    }
+  }
+  return false;
+};
+
 Room.prototype.handleMarket = function() {
-  if (!this.terminal) {
+  if (!this.terminal || this.terminal.cooldown || this.terminal.cooldown > 0) {
```

Comment:
`if (!this.terminal || this.terminal.cooldown)` should be fine, if `this.terminal.cooldown` is 0, it is a `false`.

Reply:
sure, changed